packages/react-vapor/src/components/table-hoc/examples/TableHOCServerExample.tsx

```diff
@@ -0,0 +1,193 @@
+import {withServerSideProcessing} from '../../../hoc/withServerSideProcessing/withServerSideProcessing';
+import {IDispatch, IThunkAction} from '../../../utils/ReduxUtils';
+import {IReactVaporTestState} from '../../../utils/tests/TestUtils';
+import {LastUpdated} from '../../lastUpdated/LastUpdated';
+import {Section} from '../../section/Section';
+import {TableWithPaginationActions} from '../actions/TableWithPaginationActions';
+import {TableHeaderWithSort} from '../TableHeaderWithSort';
+import {TableHOC} from '../TableHOC';
+import {TableRowHeader} from '../TableRowHeader';
+import {TableRowNumberHeader} from '../TableRowNumberHeader';
+import {tableWithActions} from '../TableWithActions';
+import {tableWithBlankSlate} from '../TableWithBlankSlate';
+import {tableWithDatePicker} from '../TableWithDatePicker';
+import {tableWithFilter} from '../TableWithFilter';
+import {tableWithNewPagination} from '../TableWithNewPagination';
+import {tableWithPredicate} from '../TableWithPredicate';
+import {tableWithSort} from '../TableWithSort';
+import {tableWithUrlState} from '../TableWithUrlState';
+import {TableHOCExampleUtils} from '../utils/TableHOCExampleUtils';
+import {ITableHOCCompositeState, TableHOCUtils} from '../utils/TableHOCUtils';
+
+type TableHOCServerProps = RouteComponentProps & ReturnType<typeof mapDispatchToProps>;
+
+export interface IExampleRowData {
+    city: string;
+    email: string;
+
+    username: string;
+    dateOfBirth: Date;
+    id: string;
+}
+
+export interface IExampleServerTableState {
+    data: IExampleRowData[];
+    isLoading: boolean;
+}
+
+interface TableHOCServerExamplesState {
+    data: {users: [any]; count: number};
+    isLoading: boolean;
+}
+
+export const TableHOCServerExample: ExampleComponent = () => <TableHOCServer />;
+
+TableHOCServerExample.title = 'TableHOC server';
+
+// start-print
+export const TableHOCServerExampleId = 'complex-example';
+
+const renderHeader = (isLoading: boolean) => (
+    <thead>
+        <tr>
+            <TableRowNumberHeader isLoading={isLoading} />
+            <TableHeaderWithSort id="address.city" tableId={TableHOCServerExampleId} isLoading={isLoading}>
+                City
+            </TableHeaderWithSort>
+            <TableHeaderWithSort id="email" tableId={TableHOCServerExampleId} isLoading={isLoading}>
+                Email
+            </TableHeaderWithSort>
+            <TableHeaderWithSort id="username" tableId={TableHOCServerExampleId} isLoading={isLoading} isDefault>
+                Username
+            </TableHeaderWithSort>
+            <TableRowHeader isLoading={isLoading}>Date of Birth</TableRowHeader>
+            <TableRowHeader isLoading={isLoading} />
+        </tr>
+    </thead>
+);
+
+const mapDispatchToProps = (dispatch: IDispatch) => ({
+    fetch: () => dispatch(TableHOCServerActions.fetchData()),
+});
+
+class TableExampleDisconnected extends React.PureComponent<TableHOCServerProps, TableHOCServerExamplesState> {

```

Comment:
why reverting it to a class component ?

Reply:
to use the internal state, I can switch with a functional component with hooks but I prefer class when I have more than 1 value in the state